linkerd/server-policy/src/network.rs

```diff
@@ -0,0 +1,69 @@
+use ipnet::IpNet;
+
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct Network {
+    pub net: IpNet,
+    pub except: Vec<IpNet>,
+}
+
+// === impl Network ===
+
+impl From<IpNet> for Network {
+    fn from(net: IpNet) -> Self {
+        Self {
+            net,
+            except: vec![],
+        }
+    }
+}
+
+impl From<ipnet::Ipv4Net> for Network {
+    fn from(net: ipnet::Ipv4Net) -> Self {
+        Self::from(IpNet::from(net))
+    }
+}
+
+impl From<ipnet::Ipv6Net> for Network {
+    fn from(net: ipnet::Ipv6Net) -> Self {
+        Self::from(IpNet::from(net))
+    }
+}
+
+impl From<std::net::IpAddr> for Network {
+    fn from(addr: std::net::IpAddr) -> Self {
+        Self::from(ipnet::IpNet::from(addr))
+    }
+}
+
+impl From<std::net::Ipv4Addr> for Network {
+    fn from(addr: std::net::Ipv4Addr) -> Self {
+        Self::from(ipnet::Ipv4Net::from(addr))
+    }
+}
+
+impl From<std::net::Ipv6Addr> for Network {
+    fn from(addr: std::net::Ipv6Addr) -> Self {
+        Self::from(ipnet::Ipv6Net::from(addr))
+    }
+}
+
+impl Network {
+    #[inline]
+    pub fn contains(&self, addr: &std::net::IpAddr) -> bool {
+        let contains_ip = self.net.contains(addr);
+        let exception = self.except.iter().any(|net| net.contains(addr));
+        contains_ip && !exception
```

Comment:
nit/tioli: if this was 
```suggestion
        contains_ip && self.except.iter().any(|net| net.contains(addr))
```
or something, instead of using a variable, we could avoid doing the iteration over `self.except` if the address is not contained in `self.net`. the way this is written now, we will *always* check if there's an exception, even if we don't need to.

not sure if this is a big enough overhead to actually matter, but felt like it was worth pointing out...